_posts/2020-06-13-IrishGen-RDF-Linked-Data-Semantic-Web.md

```diff
@@ -0,0 +1,398 @@
+Triplestore.  There are many open source and commercial Triplestores
+in use but IrishGen is generally used with
+[GraphDB](https://www.ontotext.com/products/graphdb/) or
+[Stardog](https://www.stardog.com/).  Querying RDF is a deep and
+complicated subject that will be reserved for another post where the
+implications can be properly explored in depth.
+
+The foregoing demonstrate in a microcosm how the curators of IrishGen
+use RDF and its seralisations to organise medieval Irish geneaological
+information.  This, of course, is just one small sample of the
+information avaliable within IrishGen.  However, this should orient
+and assist the reader when they encounter the IrishGen in its file
+format.  The rest of this post will be devoted to the various kinds of
+errors and difficulies that a reader and curator can encounter when
+using the dataset.
+
+## Difficulties and Challenges of Using RDF
+
+Now the reader is aquainted with the various parts of RDF and its
+ecosystem of terms.  The choices made do not come without cost and
+without their own challenges.  This section will explore some of the
+choices that IrishGen has encountered over the years and give some
+solutions to those which may be of interest to readers.
+
+Human error is the bane of may digital curation projects.  IrishGen is
+no less effected.  While typos are generally an annoyance, in IrishGen
+and RDF generally, they can cause entire systems to go awry.  A case
+in point, a vast number of the entries in IrishGen have the form
+below:
+
+```turtle
+<#Flaithbertach>
+    a foaf:Person;
+    irishRel:nomName "Flaithbertach";
+    rel:childOf <#Crunmael>.
+```
+
+Informally, this states that Flaithbertach is a person who has the
+name (in the nominative case) Flaithbertach and is the child of a
+person denoted by the URL `<#Crunmael>`.  While automated generation
+of this entry can help (as discussed in [Human Curation and Digital
+Datasets: A Problem in Multiple
+Parts]({% post_url 2020-06-13-IrishGen-RDF-Linked-Data-Semantic-Web %})),
+this still requires human intervention and thus the possibility of
+error.  If an error is introduced of the form:
+
+```
+<#Flaithbertach>
+    a foaf:Person;
+    irishRel:nomName "Flaithbertach";
+    rel:childOF <#Crunmael>.
+
+```
+
+Notice the difference between `childOf` and `childOF`.  This is subtle
+and easily missed even during a full audit.  The effect of this change
+is that the predicate is assumed to exist but it has no effect.
+Essentially, this means that when placed in a Triplestore that can
+apply reasoning to a triple, no reasoning will apply.  The implication
+of this is that now Flaithbertach is not the child of Crunmael and
+when searched for under its proper form, `childOf`, it will not
+appear.  The system is broken and if this is in a complex query,
+incorrect results will be returned to the user.  The old adage
+"Garbage In, Garbage Out" but it is more serious because given the
```

Comment:
Something seems to be missing here. "The old adage "Garbage In, Garbage Out" but" > "The old adage
"Garbage In, Garbage Out" still applies but"?